crates/ibc/src/events.rs

```diff
@@ -229,6 +296,9 @@ pub enum IbcEvent {
     ChannelClosed(ChannelEvents::ChannelClosed),
 
     AppModule(ModuleEvent),
+
+    Message(IbcEventType),
+    ClientMisbehaviourMessage(Signer),
```

Comment:
I'd model this differently, but exactly how will depend on the outcome of the other comments